File: 0128-longest-consecutive-sequence/0128-longest-consecutive-sequence.cpp
class Solution {
public:
//optimized using set ds
    int longestConsecutive(vector<int>& nums) {
        int longest=0;
        unordered_set<int>st;
        //insert elements in the set
        for(int i=0;i<nums.size();i++){
            st.insert(nums[i]);
        }
        //iterate through the set
        for(auto it:st){
            if(st.find(it-1)==st.end()){
                int cnt=1;
                int x=it;
                while(st.find(x+1)!=st.end()){
                    x=x+1;
                    cnt++;
                }
            longest=max(longest,cnt);
            }
        }
        return longest;
    }
};

File: 0128-longest-consecutive-sequence/0128-longest-consecutive-sequence.java
class Solution {
    public int longestConsecutive(int[] nums) {
        int longest=1;
        if(nums.length==0)return 0;
        Set<Integer>st=new HashSet<>();
        for(int i=0;i<nums.length;i++){
            st.add(nums[i]);
        }
        for(int x:st){
            if(!st.contains(x-1)){
                int cnt=1;
                int curr=x;
            while(st.contains(curr+1)){
                    cnt++;
                    curr=curr+1;
                }
            longest=Math.max(longest,cnt);
            }
        }
        return longest;

    }
}